### ISSUE TITLE
instanceof keyword not allowed in Exercise 4

### ISSUE THREAD

**Issue by limyeechern** [nus-cs2030/2223-s1#75]
Hi, the keyword instanceof is disallowed in Exercise 4 although the equals method that we are taught in recitation involved the use of instanceof. Is there a way to allow instanceof keyword only in this equals method for this exercise? Otherwise we might need to typecast to implement the equals method, which Prof does not like.

<img width="777" alt="image" src="https://user-images.githubusercontent.com/66681646/190214544-4bb044b1-c32d-4c27-a5b6-18b837c4dec4.png">

**Comment by limyeechern** [comment]
> Can I ask what are you using the instanceof keyword for in Exercise 4? I didn't use it and still managed to get the answer, so you may want to consider changing your implementation.

I used instanceof in the equals method, according to the way equals method is implemented in recitation 2 or 3

**Comment by junhuiii** [comment]
Why would you need to define an equals method for this exercise?

**Comment by limyeechern** [comment]
> Why would you need to define an equals method for this exercise?

The solution I thought of to check if the puzzle is solved is it check if the puzzle is equals to a new puzzle of same dimensions, which will be in the correct/solved order.

**Comment by junhuiii** [comment]
Ahh ok understood, I manually went to check if the positions are the same, hence I didn't write a .equals() method. 

Don't think there's anything wrong your implementation honestly, just that CodeCrunch doesn't want you to use instanceof....maybe try removing it for this exercise, since there is isn't any inheritance at play so to speak?

**Comment by bryanolm99** [comment]
You can simply compare the output of toString() to see if the puzzles are the same :)

**Comment by wilsonwid** [comment]
I don't think you need to define an `equals()` method as you can manually compare elements manually. Likewise, comparing the output of the `toString()` works too :")

**Comment by ccheemeng** [comment]
To add on to Bryan's suggestion, if you wish to use the equals method you can use the String.equals() method to compare the .toString() outputs of the puzzle and the solution.
